src/sessions/SessionManager.ts

```diff
@@ -81,10 +82,12 @@ export class SessionManager {
         this.record = record;
         this.pageManager = pageManager;
 
+        this.isSampled = this.sample();
```

Comment:
issue: This changes the behavior so that the sampling decision persists for all sessions. The correct behavior is that every session is sampled randomly.

I'm a little surprised that we don't have a unit test to cover this behavior.

Reply:
Refactored and added unit test.